bot/seasons/evergreen/space.py

```diff
@@ -0,0 +1,217 @@
+
+    async def convert(self, ctx: Context, argument: str) -> Union[int, datetime]:
+        """Parse date (SOL or earth) into `datetime` or `int`. When invalid value, raise error."""
+        if argument.isdigit():
+            return int(argument)
+        try:
+            date = datetime.strptime(argument, "%Y-%m-%d")
+        except ValueError:
+            raise BadArgument(f"Can't convert `{argument}` to `datetime` in format `YYYY-MM-DD` or `int` in SOL.")
+        return date
+
+
+class Space(Cog):
+    """Space Cog contains commands, that show images, facts or other information about space."""
+
+    def __init__(self, bot: SeasonalBot):
+        self.bot = bot
+        self.http_session = bot.http_session
+
+        self.rovers = {}
+        self.get_rovers.start()
+
+    @tasks.loop(hours=24)
+    async def get_rovers(self) -> None:
+        """Get listing of rovers from NASA API and info about their start and end dates."""
+        data = await self.fetch_from_nasa("mars-photos/api/v1/rovers", params={"api_key": Tokens.nasa})
+
+        for rover in data["rovers"]:
+            self.rovers[rover["name"].lower()] = {
+                "min_date": rover["landing_date"],
+                "max_date": rover["max_date"]
+            }
+
+    @group(name="space", invoke_without_command=True)
+    async def space(self, ctx: Context) -> None:
+        """Head command that contains commands about space."""
+        await ctx.send_help("space")
+
+    @space.command(name="apod")
+    async def apod(self, ctx: Context, date: Optional[str] = None) -> None:
+        """
+        Get Astronomy Picture of Day from NASA API. Date is optional parameter, what formatting is YYYY-MM-DD.
+
+        If date is not specified, this will get today APOD.
+        """
+        params = APOD_DEFAULT_PARAMS.copy()
+        # Parse date to params, when provided. Show error message when invalid formatting
+        if date:
+            try:
+                params["date"] = datetime.strptime(date, "%Y-%m-%d").date().isoformat()
+            except ValueError:
+                await ctx.send(f"Invalid date {date}. Please make sure your date is in format YYYY-MM-DD.")
+                return
+
+        result = await self.fetch_from_nasa("planetary/apod", params)
+
+        await ctx.send(embed=await self.create_nasa_embed(
+            f"Astronomy Picture of the Day - {result['date']}",
+            result["explanation"],
+            result["url"]
+        ))
+
+    @space.command(name="nasa")
+    async def nasa(self, ctx: Context, *, search_term: Optional[str] = None) -> None:
+        """Get random NASA information/facts + image. Support `search_term` parameter for more specific search."""
+        params = {
+            "media_type": "image"
+        }
+        if search_term:
+            params["q"] = search_term
+
+        data = await self.fetch_from_nasa("search", params, NASA_IMAGES_BASE_URL)
```

Comment:
There needs to be a comment stating that an API key isn't needed for this endpoint.